### DISCUSSION TITLE
Is it also possible to calculate the surface temperature of building walls using SOLWEIG?

### DISCUSSION THREAD

**Discussion by kg85112** [UMEP-dev/UMEP#593] [Q&A] [answered]
Hello, my name is G.Kang, and I work in the field of urban planning.

I recently started using SOLWEIG to calculate MRT, which has been quite exciting.

However, for my 3D CFD simulations, I need the surface temperatures of not only the ground but also the building walls.

With SOLWEIG, I understand that the surface temperature or MRT can be calculated for the ground or roofs, but is there no output for wall temperatures?

Is there any way to calculate the temperature of the walls?

Thank you very much!

![image](https://github.com/UMEP-dev/UMEP/assets/164312008/90b3f3c4-f9fb-459e-92a0-7c4cf6942fe4)

**Comment by biglimp** [comment, marked as answer]
Hi. At the moment, we dont have wall temperature as an output but we are soon ready to release a new version where this is possible. I you could wait we will soon present SOLWEIG 2024a which has these opportunities.

**Reply by kg85112** [reply]
Thank you very much for your prompt reply. I look forward to the release of SOLWEIG 2024a and the new opportunities it will bring. 
Thank you again for your support and for developing such valuable tools for the community.

**Comment by Naeim-Mj** [comment]
> Hello, my name is G.Kang, and I work in the field of urban planning.
> 
> I recently started using SOLWEIG to calculate MRT, which has been quite exciting.
> 
> However, for my 3D CFD simulations, I need the surface temperatures of not only the ground but also the building walls.
> 
> With SOLWEIG, I understand that the surface temperature or MRT can be calculated for the ground or roofs, but is there no output for wall temperatures?
> 
> Is there any way to calculate the temperature of the walls?
> 
> Thank you very much!
> 
> ![image](https://private-user-images.githubusercontent.com/164312008/315140170-90b3f3c4-f9fb-459e-92a0-7c4cf6942fe4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzEwMjUxOTYsIm5iZiI6MTc3MTAyNDg5NiwicGF0aCI6Ii8xNjQzMTIwMDgvMzE1MTQwMTcwLTkwYjNmM2M0LWY5ZmItNDU5ZS05MmEwLTdjNGNmNjk0MmZlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIxM1QyMzIxMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MzBlMmU1ZTU4YzI2Y2Q2MzZhMzcyYTg4ZGMwOTIwNGY2ZGM4YTc3ODZlOWM2ZmZhZWI4ZDM1YTZkYmJiMzNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CtIrwhAOtp4FGJUqOtzB-K_kuL6hJQb2BlzNUvo7uas)

Hi @kg85112, 

I came across your message about calculating wall surface temperatures. I’m working on a similar problem and was wondering if you were able to find a way to compute or extract wall temperatures.

Did you manage to get any output for wall surface temperatures, or find an alternative approach?

Thanks in advance!
Naeim

**Reply by kg85112** [reply]
Hi there. In my case, I modified the source code to directly extract wall temperatures. Specifically, I focused on:
 1) Determining whether walls were shaded by buildings or vegetation.
 2) Fitting sunlit and shaded temperatures using measured data.

However, from what I know, the new version (v2025a) now supports wall temperature calculations. Since I haven't used v2025a myself yet, I’m not entirely sure, but it might be worth checking out.

**Reply by biglimp** [reply]
Hi, The new wall surface scheme is now included. in the development release soon to be published in the QGIS repo. Here is the publication: https://doi.org/10.5194/gmd-19-1321-2026.

**Reply by Naeim-Mj** [reply]
> Hi there. In my case, I modified the source code to directly extract wall temperatures. Specifically, I focused on:
> 
> 1. Determining whether walls were shaded by buildings or vegetation.
> 2. Fitting sunlit and shaded temperatures using measured data.
> 
> However, from what I know, the new version (v2025a) now supports wall temperature calculations. Since I haven't used v2025a myself yet, I’m not entirely sure, but it might be worth checking out.

Hi @kg85112,
Thank you for your response. I really appreciate it.

Yes, I’ve worked with the new version (v2025a) and some of the provided test datasets (https://zenodo.org/records/15355718), but I’m not fully confident about the outputs I’m getting. If possible, could you share your improved code or the test datasets you used? I’d like to run the model with those as well.

Thanks for your response, and thanks in advance for any materials you can share.

Best regards,
Naeim
naeim.m992@gmail.com

**Reply by biglimp** [reply]
@nilswallenberg , is planning for a tutorial on the wall scheme. The zenodo is related to the paper and have some tweaks, such as hardcoded estimation of direct solar radiation and some other small things. We suggest that you try is out using the development release and e.g. the standard Gothenburg Dataset used in the other SOLWEIG tutorials online.

**Reply by Naeim-Mj** [reply]
> @nilswallenberg , is planning for a tutorial on the wall scheme. The zenodo is related to the paper and have some tweaks, such as hardcoded estimation of direct solar radiation and some other small things. We suggest that you try is out using the development release and e.g. the standard Gothenburg Dataset used in the other SOLWEIG tutorials online.

Hi @biglimp,
Thank you very much for taking the time to respond to my message. I truly appreciate your helpful explanations and valuable guidance.

I just have a quick follow up question. Where can I access or download the development version you mentioned?
Additionally, I was wondering if you might have an approximate timeline for when the official tutorial on the wall scheme will be available. I completely understand that an exact date may not be possible; I am just hoping to get a general idea of when it might be ready.

Thank you again for your support.
Best regards,
Naeim

**Reply by biglimp** [reply]
We have just uploaded to QGIS repo so you can access the latest version via QGIS now. Regarding the tutorial I would guess before the summer...

**Reply by kg85112** [reply]
> > Hi there. In my case, I modified the source code to directly extract wall temperatures. Specifically, I focused on:
> > 
> > 1. Determining whether walls were shaded by buildings or vegetation.
> > 2. Fitting sunlit and shaded temperatures using measured data.
> > 
> > However, from what I know, the new version (v2025a) now supports wall temperature calculations. Since I haven't used v2025a myself yet, I’m not entirely sure, but it might be worth checking out.
> 
> Hi @kg85112, Thank you for your response. I really appreciate it.
> 
> Yes, I’ve worked with the new version (v2025a) and some of the provided test datasets (https://zenodo.org/records/15355718), but I’m not fully confident about the outputs I’m getting. If possible, could you share your improved code or the test datasets you used? I’d like to run the model with those as well.
> 
> Thanks for your response, and thanks in advance for any materials you can share.
> 
> Best regards, Naeim [naeim.m992@gmail.com](mailto:naeim.m992@gmail.com)

Sadly, I lost my QGIS code tweaks when the plugin updated. Go ahead and try biglimp's suggestion for now. I’ll get the code back up and running soon, but I can't share it just yet.

I expect to be able to share it by early April at the earliest.
